code/game/objects/structures/crates_lockers/closets.dm

```diff
@@ -183,22 +199,132 @@
 		return src.close()
 	return src.open()
 
+/obj/structure/closet/proc/add_lock(var/obj/item/weapon/circuitboard/airlock/E, var/mob/user)
+	if(has_lock_type && !electronics && E && E.icon_state != "door_electronics_smoked")
+		playsound(src, 'sound/items/Screwdriver.ogg', 100, 1)
+		user.visible_message("[user] is installing electronics on \the [src].", "You start to install electronics into \the [src].")
+		if(do_after(user, src, 40))
+			var/obj/structure/closet/new_closet
+			new_closet = change_type(has_lock_type)
+
+			if(new_closet)
+				to_chat(user, "<span class='notice'>You installed the electronics!</span>")
+				user.drop_item(E, new_closet, force_drop = 1)
```

Comment:
why force_drop? What if a clown superglued the electronics? Check for the result of `drop_item` instead.

Reply:
I copied that from airlock building code iirc. Testing it out, can drop superglued electronics into airlock assembly frames.